pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyTest.java

```diff
@@ -429,6 +436,63 @@ public void testGetClientVersion() throws Exception {
+        final String topic = BrokerTestUtil.newUniqueName("persistent://public/default/tp");
+        admin.topics().createNonPartitionedTopic(topic);
+
+        // Create a client as a old version, which does not support "supportsReplDedupByLidAndEid".
+        ClientBuilderImpl clientBuilder2 =
+                (ClientBuilderImpl) PulsarClient.builder().serviceUrl(proxyService.getServiceUrl());
+        PulsarClientImpl injectedClient = InjectedClientCnxClientBuilder.create(clientBuilder2,
+            (conf, eventLoopGroup) -> {
+                return new ClientCnx(InstrumentProvider.NOOP, conf, eventLoopGroup) {
+
+                    @Override
+                    protected ByteBuf newConnectCommand() throws Exception {
+                        authenticationDataProvider = authentication.getAuthData(remoteHostName);
+                        AuthData authData = authenticationDataProvider.authenticate(AuthData.INIT_AUTH_DATA);
+                        BaseCommand cmd =
+                                Commands.newConnectWithoutSerialize(authentication.getAuthMethodName(), authData,
+                                        this.protocolVersion, clientVersion, proxyToTargetBrokerAddress,
+                                        null, null, null, null, null);
+                        cmd.getConnect().getFeatureFlags().setSupportsAuthRefresh(supported);
+                        cmd.getConnect().getFeatureFlags().setSupportsBrokerEntryMetadata(supported);
+                        cmd.getConnect().getFeatureFlags().setSupportsPartialProducer(supported);
+                        cmd.getConnect().getFeatureFlags().setSupportsTopicWatchers(supported);
+                        cmd.getConnect().getFeatureFlags().setSupportsReplDedupByLidAndEid(supported);
+                        cmd.getConnect().getFeatureFlags()
+                                .setSupportsGetPartitionedMetadataWithoutAutoCreation(supported);
+                        return Commands.serializeWithSize(cmd);
+                    }
+                };
+            });
+
+        // Verify: the broker will create a connection, which disabled "supportsReplDedupByLidAndEid".
+        Producer<byte[]> producer = injectedClient.newProducer().topic(topic).create();
+        ServerCnx serverCnx = (ServerCnx) pulsar.getBrokerService().getTopic(topic, false).get().get()
+                .getProducers().values().iterator().next().getCnx();
+        assertEquals(serverCnx.getFeatures().isSupportsAuthRefresh(), supported);
+        assertEquals(serverCnx.getFeatures().isSupportsBrokerEntryMetadata(), supported);
+        assertEquals(serverCnx.getFeatures().isSupportsPartialProducer(), supported);
+        assertEquals(serverCnx.getFeatures().isSupportsTopicWatchers(), supported);
+        assertEquals(serverCnx.getFeatures().isSupportsReplDedupByLidAndEid(), supported);
+        assertEquals(serverCnx.getFeatures().isSupportsGetPartitionedMetadataWithoutAutoCreation(), supported);
```

Comment:
use a local variable for `serverCnx.getFeatures()` to improve code readability.

Reply:
done